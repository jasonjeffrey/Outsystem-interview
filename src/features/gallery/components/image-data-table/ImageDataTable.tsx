import React, {FunctionComponent} from 'react';
import {Table, TableBody, TableCell, TableContainer, TableRow} from '@material-ui/core';

type ImageDataTableProps = {
    url?: string;
    width?: number;
    height?: number;
    xResolution?: number;
    yResolution?: number;
}

export const ImageDataTable: FunctionComponent<ImageDataTableProps> = ({url, width, height, xResolution, yResolution}) => (
    <TableContainer>
        <Table>
            <TableBody>
                <TableRow>
                    <TableCell>Url</TableCell>
                    <TableCell>{url}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell>Width</TableCell>
                    <TableCell>{width}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell>Height</TableCell>
                    <TableCell>{height}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell>XResolution</TableCell>
                    <TableCell>{xResolution}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell>YResolution</TableCell>
                    <TableCell>{yResolution}</TableCell>
                </TableRow>
            </TableBody>
        </Table>
    </TableContainer>
);
