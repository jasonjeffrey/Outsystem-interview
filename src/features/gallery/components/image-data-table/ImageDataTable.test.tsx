import {render} from '@testing-library/react';
import {ImageDataTable} from './ImageDataTable';

describe('ImageDataTable Component', () => {
    it('RENDERS component without any filled props', () => {
        const component = <ImageDataTable />
        const {asFragment} = render(component);

        expect(asFragment()).toMatchSnapshot();
    });

    it('RENDERS component with filled props', () => {
        const component = <ImageDataTable  url={'http://example.com'}
                                           height={768}
                                           width={1080}
                                           xResolution={72}
                                           yResolution={72}/>
        const {asFragment} = render(component);

        expect(asFragment()).toMatchSnapshot();
    });
});
