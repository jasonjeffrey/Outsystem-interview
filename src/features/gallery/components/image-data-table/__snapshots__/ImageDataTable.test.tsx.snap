// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDataTable Component RENDERS component with filled props 1`] = `
<DocumentFragment>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Url
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            http://example.com
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Width
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1080
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Height
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            768
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            XResolution
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            72
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            YResolution
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            72
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`ImageDataTable Component RENDERS component without any filled props 1`] = `
<DocumentFragment>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Url
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          />
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Width
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          />
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            Height
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          />
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            XResolution
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          />
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            YResolution
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
